client/components/codeEditor.jsx

```diff
@@ -22,18 +27,72 @@ export default function CodeEditor(props) {
   }, [html, css, javascript]);
 
   function addtoDatabase() {
-    const newObject = { html, css, javascript };
+    const newObject = { html, css, javascript, title, imageUrl, description };
     fetch('/api/code', { method: 'POST', body: JSON.stringify(newObject), headers: { 'Content-Type': 'application/json' } });
+    setUrlOnChange(pre => false);
+  }
+  function handleTitle(event) {
+    setTitle(event.target.value);
+  }
+  function handleImage(event) {
+    setImageUrl(event.target.value);
+    setUrlOnChange(pre => true);
+  }
```

Comment:
Instead of two states for image url, I recommend using only one state. You can make the start state be the placeholder.